using_images/db_images/mongodb.adoc

```diff
@@ -456,54 +456,76 @@ of the *rs0* replication set, as the value of `*MONGODB_REPLICA_NAME*` is set to
 than a regular, stand-alone MongoDB database.
 
-[[example-service-pod]]
-.Service Pod Object Definition in the Example Template
+[[example-deployment-config]]
+.Deployment Configuration Object Definition in the Example Template
 ====
 
 [source,yaml]
 ----
-kind: "Pod"
-apiVersion: "v1"
-metadata:
-  name: "mongodb-service"
-  creationTimestamp: null
-  labels:
-    name: "mongodb-service"
-spec:
-  restartPolicy: "Never"
-  dnsPolicy: "ClusterFirst"
-  containers:
-    - name: "initiate"
-      image: "centos/mongodb-26-centos7"
-      args:
-        - "initiate"
-      env:
-        - name: "MONGODB_USER"
-          value: "${MONGODB_USER}"
-        - name: "MONGODB_PASSWORD"
-          value: "${MONGODB_PASSWORD}"
-        - name: "MONGODB_DATABASE"
-          value: "${MONGODB_DATABASE}"
-        - name: "MONGODB_ADMIN_PASSWORD"
-          value: "${MONGODB_ADMIN_PASSWORD}"
-        - name: "MONGODB_REPLICA_NAME"
-          value: "${MONGODB_REPLICA_NAME}"
-        - name: "MONGODB_SERVICE_NAME"
-          value: "${MONGODB_SERVICE_NAME}"
-        - name: "MONGODB_KEYFILE_VALUE"
-          value: "${MONGODB_KEYFILE_VALUE}"
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: "${MONGODB_SERVICE_NAME}"
+  spec:
+    strategy:
+      type: Recreate
+      recreateParams:
+        post:
+          failurePolicy: Retry
+          execNewPod:
+            command:
+            - run-mongod
+            - initiate
+            containerName: mongodb
+            env:
+            - name: MONGODB_INITIAL_REPLICA_COUNT
+              value: '3'
+    triggers:
+    - type: ConfigChange
+    replicas: 3
+    selector:
+      name: mongodb-replica
+    template:
+      metadata:
+        labels:
+          name: mongodb-replica
+      spec:
+        containers:
+        - name: mongodb
+          image: openshift/mongodb-24-centos7
+          readinessProbe:
+            tcpSocket:
+              port: 27017
+            initialDelaySeconds: 15
+            timeoutSeconds: 1
+          env:
+          - name: MONGODB_USER
+            value: "${MONGODB_USER}"
+          - name: MONGODB_PASSWORD
+            value: "${MONGODB_PASSWORD}"
+          - name: MONGODB_DATABASE
+            value: "${MONGODB_DATABASE}"
+          - name: MONGODB_ADMIN_PASSWORD
+            value: "${MONGODB_ADMIN_PASSWORD}"
+          - name: MONGODB_REPLICA_NAME
+            value: "${MONGODB_REPLICA_NAME}"
+          - name: MONGODB_SERVICE_NAME
+            value: "${MONGODB_SERVICE_NAME}"
+          - name: MONGODB_KEYFILE_VALUE
+            value: "${MONGODB_KEYFILE_VALUE}"
+          ports:
+          - containerPort: 27017
 ----
 ====
 
 [[mongo-creating-headless-service]]
 === Creating a Headless Service
 
-The `initiate` argument in the xref:example-service-pod[container specification
+The `initiate` argument in the xref:example-deployment-config[container specification
```

Comment:
Probably more chunks of this prose don't reflect the current example template...

I find little value in this docs talking about "creating a headless service" or describing in too much specificity how the replication example is implemented, as it easily rots.

We may well stop using headless services, or change the exact steps for initializing the replica set.